var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -13,41 +13,83 @@ class Elk(MakefilePackage):
     homepage = 'http://elk.sourceforge.net/'
     url      = 'https://sourceforge.net/projects/elk/files/elk-3.3.17.tgz'
 
-    version('3.3.17', sha256='c9b87ae4ef367ed43afc2d43eb961745668e40670995e8e24c13db41b7e85d73')
```

Comment:
Can you keep the old version for now? If you want to remove it in the future you can add `deprecated=True` and remove it in the next Spack release.

Reply:
Are you adamant it's needed? I'd rather not. It links against old libxc with different API. Consequently, it would require special handling of dependencies and/or ad-hoc patches. Besides, because of the changes in libxc API and, hm, unconstrained dependencies it didn't compile anyway.

Reply:
In the past we've simply removed them, but we're trying to be more stable and have fewer unexpected changes. See https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions

Reply:
Although it doesn't compile for you on your system, it may compile for someone else on an older system. The package was working at one time for me, and I don't think it's been changed much since then.

Reply:
There are a few changes between 3.* and 7.* . In particular,
- Changed integration with libxc (3.* uses libxc@:4 API, 7.* integrates with libxc@5: . The APIs are not compatible)
- BLISS support was added, so bliss variant doesn't make sence for elk 3.*
- (optional) supporting code for control of multithreading in MKL, fftw and OpenBlas was added.

This means that supporting 3.* would require some directives with checks on version. I really don't want to deal with this and I doubt I can do it right at my current understanding of python.

Reply:
@permeakra If you're comfortable with a Spack maintainer pushing to your PR, we can take care of separating the new logic from the logic to support the old version. The changes would be:

1. keep both edit methods. Use a `@when('@:4')` decorator on the old one so Spack knows to only call it when using the old version of the spec.
2. change `depends_on('libxc@5:', when='+libxc')` to two lines: `depends_on('libxc@:4', when='@:3.99+libsc')` and `depends_on('libxc@5:', when='@7:+libxc')`.
3. add `conflicts('+blis', when='@:3.99')` to make clear that blis support doesn't work for elk@3.
4. add similar conflicts for any other variants that don't make sense in version 3.

I think that would allow us to keep the old support around for anyone else who was using it, while moving the new support forward

Reply:
I'll push one more commit a day or two later after I ensure that a proper multihreading model is requested from the dependencies. I would be glad if a Spack maintainer took over this PR after that.